### PR TITLE
Added REORGANIZE PARTITION

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/11385
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -705,7 +705,12 @@ Empty set (0.00 sec)
 
 ## 分区管理
 
-对于 `LIST` 和 `RANGE` 分区表，通过 `ALTER TABLE <表名> ADD PARTITION (<分区说明>)` 或 `ALTER TABLE <表名> DROP PARTITION <分区列表>` 语句，可以执行添加和删除分区的操作。
+对于 `LIST` 和 `RANGE` 分区表，你可以进行以下分区管理操作：
+
+- 使用 `ALTER TABLE <表名> ADD PARTITION (<分区说明>)` 语句添加分区。
+- 使用 `ALTER TABLE <表名> DROP PARTITION <分区列表>` 删除分区。
+- 使用 `ALTER TABLE <表名> TRUNCATE PARTITION <分区列表>` 语句清空分区里的数据。`TRUNCATE PARTITION`的逻辑与 [`TRUNCATE TABLE`](/sql-statements/sql-statement-truncate.md) 相似，但它的操作对象为分区。
+- 使用`ALTER TABLE <表名> REORGANIZE PARTITION <分区列表> INTO (<新的分区说明>)`语句对分区进行合并、拆分、或者其他修改。
```
Comment (qiancai) [inline partitioned-table.md:713]:
```suggestion
- 使用 `ALTER TABLE <表名> REORGANIZE PARTITION <分区列表> INTO (<新的分区说明>)`语句对分区进行合并、拆分、或者其他修改。
```

File: partitioned-table.md [hunk 2]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
 - TiCDC：分区表和非分区表都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
 - TiDB Lightning 和 BR：使用 TiDB Lightning 导入或使用 BR 恢复的过程中，不要执行 `EXCHANGE PARTITION` 操作。
 
-### Range 分区管理
-
-创建分区表：
+### Range 分区和 List 分区管理
 
-{{< copyable "sql" >}}
+本小节将以如下 SQL 语句创建的分区表为例，介绍如何进行 Range 分区和 List 分区管理。
```
Comment (qiancai) [inline partitioned-table.md:739]:
```suggestion
本小节将以如下 SQL 语句创建的分区表为例，介绍如何管理 Range 分区和 List 分区。
```

File: partitioned-table.md [hunk 3]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+### 添加分区
 
-{{< copyable "sql" >}}
+```sql
+ALTER TABLE members ADD PARTITION (PARTITION `p1990to2010` VALUES LESS THAN (2010));
+
+ALTER TABLE member_level ADD PARTITION (PARTITION l5_6 VALUES IN (5,6));
+```
+
+对于 Range 分区，`ADD PARTITION` 只能在分区列表的最后添加新的分区。与分区列表中已有的分区相比，你需要将新分区的 `VALUES LESS THAN` 定义为更大的值。否则，执行该语句时将会报错。
```
Comment (qiancai) [inline partitioned-table.md:794]:
```suggestion
对于 Range 分区表，`ADD PARTITION` 只能在分区列表的最后添加新的分区。与分区列表中已有的分区相比，你需要将新分区的 `VALUES LESS THAN` 定义为更大的值。否则，执行该语句时将会报错。
```

File: partitioned-table.md [hunk 4]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
-    PARTITION n VALUES LESS THAN (1970));
-```
+ALTER TABLE members REORGANIZE PARTITION pBefore1950,p1950 INTO (PARTITION pBefore1960 VALUES LESS THAN (1960));
 
+ALTER TABLE member_level REORGANIZE PARTITION l1,l2 INTO (PARTITION l1_2 VALUES IN (1,2));
 ```
-ERROR 1463 (HY000): VALUES LESS THAN value must be strictly »
-   increasing for each partition
+
+修改分区定义：
```
Comment (qiancai) [inline partitioned-table.md:829]:
```suggestion
修改分区表定义：
```

File: partitioned-table.md [hunk 5]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+ALTER TABLE member_level REORGANIZE PARTITION l1_2,l3,l4,l5,l6 INTO
+(PARTITION lOdd VALUES IN (1,3,5),
+ PARTITION lEven VALUES IN (2,4,6));
 ```
 
+在重组分区时，需要注意以下关键点：
+
+- 重组分区（包括合并或拆分分区）只能修改分区定义，无法修改分区表类型。例如，无法将 List 类型修改为 Range 类型，或将 Range COLUMNS 类型修改为 Range 类型。
+
+- 对于 Range 分区表，你只能重组表中相邻的分区：
```
Comment (qiancai) [inline partitioned-table.md:846]:
```suggestion
- 对于 Range 分区表，你只能对表中相邻的分区进行重组：
```

File: partitioned-table.md [hunk 6]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+    INSERT INTO members VALUES (313, "John", "Doe", "2022-11-22", NULL);
+    ALTER TABLE members REORGANIZE PARTITION p2000 INTO (PARTITION p2000 VALUES LESS THAN (2050)); -- 执行成功，因为 2050 包含了现有的所有行
+    ALTER TABLE members REORGANIZE PARTITION p2000 INTO (PARTITION p2000 VALUES LESS THAN (2020)); -- 执行失败，因为 2022 将对应不到分区
+    ```
+
+    ```
+    ERROR 1526 (HY000): Table has no partition for value 2022
+    ```
+
+- 对于 List 分区表，当修改分区定义中的数据集合时，必须保证新的数据集合能覆盖到该分区中现有的所有值，否则 TiDB 将返回报错。
```
Comment (qiancai) [inline partitioned-table.md:868]:
```suggestion
- 对于 List 分区表，如需修改分区定义中的数据集合，必须保证新的数据集合能覆盖到该分区中现有的所有值，否则 TiDB 将返回报错。
```

File: partitioned-table.md [hunk 7]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+
+    ```sql
+    ALTER TABLE members REORGANIZE PARTITION p1800,p2000 INTO (PARTITION p2000 VALUES LESS THAN (2100));
+    ```
+
+    ```
+    ERROR 8200 (HY000): Unsupported REORGANIZE PARTITION of RANGE; not adjacent partitions
+    ```
+
+- 对于 Range 分区表，当修改 Range 定义中的最大值时，必须保证 `VALUES LESS THAN` 中新定义的值大于现有分区中的所有值。否则，TiDB 将返回报错，提示现有的行值对应不到分区。
```
Comment (qiancai) [inline partitioned-table.md:856]:
```suggestion
- 对于 Range 分区表，如需修改 Range 定义中的最大值，必须保证 `VALUES LESS THAN` 中新定义的值大于现有分区中的所有值。否则，TiDB 将返回报错，提示现有的行值对应不到分区。
```

File: mysql-compatibility.md [hunk 8]
```diff
@@ -113,11 +113,11 @@ TiDB 中，所有支持的 DDL 变更操作都是在线执行的。与 MySQL 相
 * 不支持添加或删除 `CLUSTERED` 类型的主键。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
 * 不支持指定不同类型的索引 (`HASH|BTREE|RTREE|FULLTEXT`)。若指定了不同类型的索引，TiDB 会解析并忽略这些索引。
 * 分区表支持 `HASH`、`RANGE` 和 `LIST` 分区类型。对于不支持的分区类型，TiDB 可能会报 `Warning: Unsupported partition type %s, treat as normal table` 错误，其中 `%s` 为不支持的具体分区类型。
-* 分区表还支持 `ADD`、`DROP`、`TRUNCATE` 操作。其他分区操作会被忽略。TiDB 不支持以下分区表语法：
+* 分区表还支持 `ADD`、`DROP`、`TRUNCATE`、`REORGANIZE` 操作。其他分区操作会被忽略。TiDB 不支持以下分区表语法：
```
Comment (moyun) [inline mysql-compatibility.md:116]: > ... 操作。其他分区操作会被忽略。

here seems just need a comma ","

File: mysql-compatibility.md [hunk 9]
```diff
@@ -113,11 +113,11 @@ TiDB 中，所有支持的 DDL 变更操作都是在线执行的。与 MySQL 相
...
     + `PARTITION BY KEY`
     + `PARTITION BY LINEAR KEY`
     + `SUBPARTITION`
-    + `{CHECK|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|REORGANIZE|COALESCE} PARTITION`
+    + `{CHECK|TRUNCATE|OPTIMIZE|REPAIR|IMPORT|DISCARD|REBUILD|COALESCE} PARTITION`
 
   更多详情，请参考[分区表文档](/partitioned-table.md)。
```
Comment (moyun) [inline mysql-compatibility.md:122]: > ","

，

File: partitioned-table.md [hunk 10]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+
+    ```sql
+    ALTER TABLE members REORGANIZE PARTITION p1800,p2000 INTO (PARTITION p2000 VALUES LESS THAN (2100));
+    ```
+
+    ```
+    ERROR 8200 (HY000): Unsupported REORGANIZE PARTITION of RANGE; not adjacent partitions
+    ```
+
+- 对于 Range 分区表，如需修改 Range 定义中的最大值，必须保证 `VALUES LESS THAN` 中新定义的值大于现有分区中的所有值。否则，TiDB 将返回报错，提示现有的行值对应不到分区。
```
Comment (moyun) [inline partitioned-table.md:856]: > 否则，TiDB 将返回报错
two options:
1. 否则，TiDB 将报错
2. 否则，TiDB 将返回错误信息

Comment (moyun) [reply partitioned-table.md:856]: “返回报错” 是一病句

**Review by moyun — CHANGES_REQUESTED** [review]

File: mysql-compatibility.md (hunk 8 above)
Comment (qiancai) [inline mysql-compatibility.md:116]:
```suggestion
* 分区表还支持 `ADD`、`DROP`、`TRUNCATE`、`REORGANIZE` 操作，其他分区操作会被忽略。TiDB 不支持以下分区表语法：
```

File: partitioned-table.md (hunk 10 above)
Comment (qiancai) [inline partitioned-table.md:856]:
```suggestion
- 对于 Range 分区表，如需修改 Range 定义中的最大值，必须保证 `VALUES LESS THAN` 中新定义的值大于现有分区中的所有值。否则，TiDB 将报错，提示现有的行值对应不到分区。
```

File: partitioned-table.md [hunk 11]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+    INSERT INTO members VALUES (313, "John", "Doe", "2022-11-22", NULL);
+    ALTER TABLE members REORGANIZE PARTITION p2000 INTO (PARTITION p2000 VALUES LESS THAN (2050)); -- 执行成功，因为 2050 包含了现有的所有行
+    ALTER TABLE members REORGANIZE PARTITION p2000 INTO (PARTITION p2000 VALUES LESS THAN (2020)); -- 执行失败，因为 2022 将对应不到分区
+    ```
+
+    ```
+    ERROR 1526 (HY000): Table has no partition for value 2022
+    ```
+
+- 对于 List 分区表，如需修改分区定义中的数据集合，必须保证新的数据集合能覆盖到该分区中现有的所有值，否则 TiDB 将返回报错。
```
Comment (qiancai) [inline partitioned-table.md:868]:
```suggestion
- 对于 List 分区表，如需修改分区定义中的数据集合，必须保证新的数据集合能覆盖到该分区中现有的所有值，否则 TiDB 将报错。
```

File: mysql-compatibility.md (hunk 9 above)
Comment (qiancai) [reply mysql-compatibility.md:122]: This comma in Line 122 is correct.

File: partitioned-table.md [hunk 12]
```diff
@@ -705,7 +705,12 @@ Empty set (0.00 sec)
 
 ## 分区管理
 
-对于 `LIST` 和 `RANGE` 分区表，通过 `ALTER TABLE <表名> ADD PARTITION (<分区说明>)` 或 `ALTER TABLE <表名> DROP PARTITION <分区列表>` 语句，可以执行添加和删除分区的操作。
+对于 `LIST` 和 `RANGE` 分区表，你可以进行以下分区管理操作：
+
+- 使用 `ALTER TABLE <表名> ADD PARTITION (<分区说明>)` 语句添加分区。
+- 使用 `ALTER TABLE <表名> DROP PARTITION <分区列表>` 删除分区。
+- 使用 `ALTER TABLE <表名> TRUNCATE PARTITION <分区列表>` 语句清空分区里的数据。`TRUNCATE PARTITION`的逻辑与 [`TRUNCATE TABLE`](/sql-statements/sql-statement-truncate.md) 相似，但它的操作对象为分区。
```
Comment (qiancai) [inline partitioned-table.md:712]:
```suggestion
- 使用 `ALTER TABLE <表名> TRUNCATE PARTITION <分区列表>` 语句清空分区里的数据。`TRUNCATE PARTITION` 的逻辑与 [`TRUNCATE TABLE`](/sql-statements/sql-statement-truncate.md) 相似，但它的操作对象为分区。
```

File: partitioned-table.md [hunk 13]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+### 清空分区
+
+```sql
+ALTER TABLE members TRUNCATE PARTITION p1980;
+
+ALTER TABLE member_level TRUNCATE PARTITION l4;
 ```
 
-清空分区：
+### 添加分区
```
Comment (hfxsd) [inline partitioned-table.md:786]: 可以把这一小节放到 删除分区 小节之前。逻辑上比较合理。

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: partitioned-table.md [hunk 14]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
 - TiCDC：分区表和非分区表都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
 - TiDB Lightning 和 BR：使用 TiDB Lightning 导入或使用 BR 恢复的过程中，不要执行 `EXCHANGE PARTITION` 操作。
 
-### Range 分区管理
-
-创建分区表：
+### Range 分区和 List 分区管理
```
Comment (hfxsd) [inline partitioned-table.md:737]:
```suggestion
### 管理 Range 分区和 List 分区
```

File: partitioned-table.md (hunk 13 above)
Comment (qiancai) [reply partitioned-table.md:786]: 看了下，这个例子中这三个之间有顺序关系，不能直接调整，
最前面的例子 L775 行建了 PARTITION p1990 VALUES LESS THAN (2000)); 分区，如果这个例子之后直接放添加分区，和 L789 行的 PARTITION `p1990to2010` 冲突，需要先通过删除分区小节删除 PARTITION p1990; 分区，才能添加 PARTITION `p1990to2010`

File: partitioned-table.md [hunk 15]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
...
+    ```sql
+    INSERT INTO member_level (id, level) values (313, 6);
+    ALTER TABLE member_level REORGANIZE PARTITION lEven INTO (PARTITION lEven VALUES IN (2,4));
+    ```
+
+    ```
+    ERROR 1526 (HY000): Table has no partition for value 6
+    ```
+
 ### Hash 分区管理
```
Comment (qiancai) [inline partitioned-table.md:879]:
```suggestion
### 管理 Hash 分区
```

File: partitioned-table.md [hunk 16]
```diff
@@ -729,88 +734,156 @@ Empty set (0.00 sec)
 - TiCDC：分区表和非分区表都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
 - TiDB Lightning 和 BR：使用 TiDB Lightning 导入或使用 BR 恢复的过程中，不要执行 `EXCHANGE PARTITION` 操作。
 
-### Range 分区管理
-
-创建分区表：
+### 管理 Range 分区和 List 分区
```
Comment (qiancai) [inline partitioned-table.md:737]:
```suggestion
### 管理 Range 分区、Range COLUMNS 分区、List 分区、List COLUMNS 分区
```

File: partitioned-table.md [hunk 17]
```diff
@@ -705,7 +705,12 @@ Empty set (0.00 sec)
 
 ## 分区管理
 
-对于 `LIST` 和 `RANGE` 分区表，通过 `ALTER TABLE <表名> ADD PARTITION (<分区说明>)` 或 `ALTER TABLE <表名> DROP PARTITION <分区列表>` 语句，可以执行添加和删除分区的操作。
+对于 `LIST` 和 `RANGE` 分区表，你可以进行以下分区管理操作：
```
Comment (qiancai) [inline partitioned-table.md:708]:
```suggestion
对于 `LIST`、`LIST COLUMNS`、`RANGE`、`RANGE COLUMNS` 分区表，你可以进行以下分区管理操作：
```

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**